### DISCUSSION TITLE
st scrollback-reflow patch

### DISCUSSION THREAD

**Discussion by Abhra00** [bakkeby/st-flexipatch#147] [Q&A] [answered]
I have patched st with scrollback and scrollback-reflow  
but the issue I am having is that when I scroll with mouse/keyboard shortcut then my prompt stays at the same position
rather than going down with scroll

**Actual terminal screen**
![Screenshot_2024-08-20-20-17-41_1920x1080](https://github.com/user-attachments/assets/2598a10e-9236-418b-a791-0c3cac9e41f1)
**When I scrollback**
![Screenshot_2024-08-20-20-17-54_1920x1080](https://github.com/user-attachments/assets/445e3588-3f44-45c4-adf3-5ed8c352ab43)

Please help me @bakkeby 
I beg you

**Comment by bakkeby** [comment]
Can you show which patches you have enabled?

In principle you shouldn't need the `SCROLLBACK_PATCH` if `REFLOW_PATCH` is used. It may be that there is a combination of patches where this doesn't work.

**Reply by Abhra00** [reply]
first of all thank you so much for replying
I have patched the relow patch on top of scrollback patch because the official page says patch on top of previous
[st-scrollback-patch](https://st.suckless.org/patches/scrollback/st-scrollback-0.9.2.diff)
[st-scrollbackreflow-patch](https://st.suckless.org/patches/scrollback/st-scrollback-reflow-0.9.2.diff)

**Comment by bakkeby** [comment]
I see. I tried a clean build applying only those two patches and I can't reproduce the issue.

In the screenshots you have provided the st window title has an icon, suggesting that you have applied more patches in addition to the two.

A few pointers.

#### 1) you may as well share your whole build at this point.

I am assuming that it is not your [st-0.9.2](https://github.com/Abhra00/suckless/tree/master/st-0.9.2) build as that looks like it is using the older columns patch.

#### 2) if you regularly commit your changes then you can use git bisect to work out what commit introduced a specific behaviour

#### 3) what shell do you use and can you reproduce this using other shells?

#### 4) can you reproduce this if you use a simple prompt?

**Reply by Abhra00** [reply]
It happens with a simple prompt too

Here is my final [st](https://github.com/Abhra00/suckless-builds)

**Reply by Abhra00** [reply]
@bakkeby 
Thank you so much for helping 
I have figured out what the issue was
the issue was with ligatures patch 
so in the draw(void) of st.c  file
before               
 xdrawcursor(cx, term.c.y, term.line[term.c.y][cx],
                                term.ocx, term.ocy, term.line[term.ocy][term.ocx],
                                term.line[term.ocy], term.col);
I have added if(term.scr == 0) by looking at your code (ofc bcz you are the god maintainer)
now the issue is resolved
once again thank you
may god bless you 🙏

**Comment by bakkeby** [comment]
That does indeed solve the issue that I noticed both with your build and with the bare st with only the st-scrollback + scrollbackreflow patches applied (the cursor being printed in the last row).

Credit goes to veltza on this one he did a lot of tweaks and fixes to make reflow work with a variety of patches. The reflow implementation in st-flexipatch is based on his work in his st-sx build:

https://github.com/veltza/st-sx

I'd recommend trying that build out especially if you want to play around with sixel graphics.

**Reply by Abhra00** [reply]
ok I will it out

**Reply by Abhra00** [reply]
@bakkeby 
Sorry for this 
But could check my updated st build
Is there any issue now

[St](https://github.com/Abhra00/suckless-builds)

**Reply by bakkeby** [reply, marked as answer]
Had a look and I can't see any issues with scrolling and the prompt appearing 👍🏻

**Reply by Abhra00** [reply]
Thank you so much 
You are such a humble guy❤️
I have ever met in linux community
